sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateViewAsSelect.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.execution
+
+import org.apache.hadoop.hive.ql.metadata.HiveUtils.{unparseIdentifier => verbose}
+
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.hive.{HiveMetastoreTypes, HiveContext}
+import org.apache.spark.sql.{AnalysisException, Row, SQLContext}
+import org.apache.spark.sql.execution.RunnableCommand
+import org.apache.spark.sql.hive.client.{HiveColumn, HiveTable}
+
+private[hive]
+case class CreateViewAsSelect(
+    tableDesc: HiveTable,
+    childSchema: Seq[Attribute],
+    allowExisting: Boolean) extends RunnableCommand {
+
+  assert(tableDesc.schema == Nil || tableDesc.schema.length == childSchema.length)
+  assert(tableDesc.viewText.isDefined)
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val hiveContext = sqlContext.asInstanceOf[HiveContext]
+    val database = tableDesc.database
+    val viewName = tableDesc.name
+    val viewText = tableDesc.viewText.get
+
+    if (hiveContext.catalog.tableExists(Seq(database, viewName))) {
+      if (allowExisting) {
+        // view already exists, will do nothing, to keep consistent with Hive
+      } else {
+        throw new AnalysisException(s"$database.$viewName already exists.")
+      }
+    } else {
+      val schema = if (tableDesc.schema == Nil) {
+        childSchema.map { attr =>
+          HiveColumn(attr.name, HiveMetastoreTypes.toMetastoreType(attr.dataType), null)
+        }
+      } else {
+        childSchema.zip(tableDesc.schema).map { case (attr, col) =>
+          HiveColumn(col.name, HiveMetastoreTypes.toMetastoreType(attr.dataType), col.comment)
+        }
+      }
+
+      val columnNames = childSchema.map(f => verbose(f.name))
```

Comment:
In Hive, `unparseIdentifier(String identifier)` will call `unparseIdentifier(String identifier, Configuration conf)` with `conf = null`. The implementation of `unparseIdentifier(String identifier, Configuration conf)` is
```
public static String unparseIdentifier(String identifier, Configuration conf) {
  // In the future, if we support arbitrary characters in
  // identifiers, then we'll need to escape any backticks
  // in identifier by doubling them up.

  // the time has come
  String qIdSupport = conf == null ? null :
    HiveConf.getVar(conf, HiveConf.ConfVars.HIVE_QUOTEDID_SUPPORT);
  if ( qIdSupport != null && !"none".equals(qIdSupport) ) {
    identifier = identifier.replaceAll("`", "``");
  }
  return "`" + identifier + "`";
}
```

It will escape a backtick with double-backtick. I guess we want to have the same behavior. Actually, can we inline that function (since it is very short)?